build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryDependenciesClient.java

```diff
@@ -217,12 +221,23 @@ private HttpResponse executeDownload(String artifactUrl, boolean isHead, Map<Str
     }
 
     public void setProperties(String urlPath, String props) throws IOException {
-        String url = ArtifactoryHttpClient.encodeUrl(urlPath + "?properties=" + props);
+        String url = ArtifactoryHttpClient.encodeUrl(urlPath + "?properties=");
+        url += DeploymentUrlUtils.buildMatrixParamsString(mapPropsString(props));
         PreemptiveHttpClient client = httpClient.getHttpClient();
         HttpPut httpPut = new HttpPut(url);
         checkNoContent(client.execute(httpPut), "Failed to set properties to '" + urlPath + "'");
     }
 
+    private ArrayListMultimap<String, String> mapPropsString(String props) throws UnsupportedEncodingException {
```

Comment:
I suggest adding some integration tests, to check set props in general, including props with special characters.
I suggest that setting props with generic upload is also tested.
See the *IntegrationTestsBase* class, which is the base class of the build-info integration tests.